scripts/gen_layer.py

```diff
@@ -0,0 +1,3723 @@
+            f.write(INSTANCE_FUNCTIONS)
+            f.write(self.generate_fill_physical_device_pnext_chain())
+            f.write(self.generate_fill_queue_family_properties_pnext_chain())
+            f.write(FORMAT_PROPERTIES_PNEXT)
+            f.write(GET_PHYSICAL_DEVICE_FEATURES_PROPERTIES_FUNCTIONS)
+            f.write(ENUMERATE_FUNCTIONS)
+            f.write(QUEUE_FAMILY_PROPERTIES_FUNCTIONS)
+            f.write(PHYSICAL_DEVICE_FORMAT_FUNCTIONS)
+            f.write(TOOL_PROPERTIES_FUNCTIONS)
+            f.write(TRANSFER_DEFINES)
+            f.write(self.generate_transfer_values())
+            f.write(TRANSFER_UNDEFINE)
+            f.write(self.generate_load_device_formats())
+            f.write(LOAD_QUEUE_FAMILY_PROPERTIES)
+            f.write(self.generate_enumerate_physical_device())
+            f.write(GET_INSTANCE_PROC_ADDR)
+            f.write(END_NAMESPACE)
+            f.write(EXPORT_FUNCTIONS)
+
+    def generate_helpers(self):
+        gen = self.generate_enum_to_string('SimulateCapabilityFlags', ('SIMULATE_API_VERSION_BIT', 'SIMULATE_FEATURES_BIT', 'SIMULATE_PROPERTIES_BIT', 'SIMULATE_EXTENSIONS_BIT', 'SIMULATE_FORMATS_BIT', 'SIMULATE_QUEUE_FAMILY_PROPERTIES_BIT'))
+        gen += self.generate_string_to_enum('SimulateCapabilityFlags', ('SIMULATE_API_VERSION_BIT', 'SIMULATE_FEATURES_BIT', 'SIMULATE_PROPERTIES_BIT', 'SIMULATE_EXTENSIONS_BIT', 'SIMULATE_FORMATS_BIT', 'SIMULATE_QUEUE_FAMILY_PROPERTIES_BIT'), 'GetSimulateCapabilitiesLog')
+        gen += self.generate_string_to_enum('DebugActionFlags', ('DEBUG_REPORT_NOTIFICATION_BIT', 'DEBUG_REPORT_WARNING_BIT', 'DEBUG_REPORT_ERROR_BIT', 'DEBUG_REPORT_DEBUG_BIT'), 'GetDebugReportsLog')
+
+        gen += self.generate_string_to_enum('VkFormatFeatureFlags', self.get_non_aliased_list(registry.bitmasks["VkFormatFeatureFlags"].bitsType.values, registry.bitmasks["VkFormatFeatureFlags"].bitsType.aliasValues), 'GetFormatFeatureString')
+        gen += self.generate_string_to_enum('VkFormatFeatureFlags2', self.get_non_aliased_list(registry.bitmasks["VkFormatFeatureFlags2"].bitsType.values, registry.bitmasks["VkFormatFeatureFlags2"].bitsType.aliasValues), 'GetFormatFeature2String')
+        gen += self.generate_string_to_enum('VkQueueFlags', registry.bitmasks["VkQueueFlags"].bitsType.values, 'GetQueueFlagsToString')
+
+        gen += self.generate_format_to_string(registry.enums['VkFormat'].values, registry.enums['VkFormat'].aliasValues)
+        gen += self.generate_string_to_format(registry.enums['VkFormat'].values)
+
+        gen += self.generate_string_to_uint(('VkSampleCountFlagBits', 'VkResolveModeFlagBits', 'VkShaderStageFlagBits', 'VkSubgroupFeatureFlagBits', 'VkShaderFloatControlsIndependence', 'VkPointClippingBehavior'), registry.enums)
+
+        gen += self.generate_string_to_flag_functions(('VkFormatFeatureFlags', 'VkQueueFlags', 'VkQueueGlobalPriorityKHR', 'VkVideoCodecOperationFlagsKHR', 'VkPipelineStageFlags', 'VkPipelineStageFlags2', 'VkFormatFeatureFlags2'))
```

Comment:
In fact, this is why I suggested that maybe we should add a reflection API generator which would generate these conversion functions and then that could have been used for the following purposes:

* Used by the profiles layer for the purpose you use it for here
* Used by the debug variant of the profiles library to print out actual values in pretty format, which is currently not supported as it would have added significant development time to the otherwise development-only use debug variant generator
* Used by `vulkaninfo` to be able to dump profiles if we would have chosen to add profile dumping utilities to the Vulkan-Profiles repo instead of changing the dev-sim generator in `vulkaninfo`

Reply:
These were the only ones used, but you're right, I should generate functions for them all. The second part will also probably be done in a future PR.

Reply:
Sure, I don't think it's necessary at this point. It was just a suggestion.